### PR TITLE
Fix SortedList<T>.IndexOf(T) returning a valid value for an item not contained by it, instead of -1.

### PR DESCRIPTION
**Pull request by smoogipooo**

### CODE REVIEW

File: osu.Framework/Graphics/Containers/CompositeDrawable.cs [hunk 1]
```diff
@@ -276,7 +277,7 @@ protected internal void ClearInternal(bool disposeChildren = true)
         /// Used to assign a monotonically increasing ID to children as they are added. This member is
         /// incremented whenever a child is added.
         /// </summary>
-        private ulong currentChildID;
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs]: I'd prefer if this stayed non-static. I don't see any particularly appealing reason to make it static. Sure, we save a bunch of bytes per container, but we also introduce an atomic operation. We should avoid statics whereever possible imho.

Comment (Tom94) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs]: Also, I'd prefer it to remain an `ulong` to capture the fact that we expect these IDs to be positive.

File: osu.Framework.Desktop.Tests/Visual/TestCaseContainerState.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        {
+            var drawableA = new Sprite();
+            var drawableB = new Sprite();
+            var containerA = new Container { Child = drawableA };
+            var containerB = new Container { Child = drawableB };
+
+            var newContainer = new Container<Container> { Children = new[] { containerA, containerB } };
+
+            // Because drawableA and drawableB have been added to separate containers,
+            // they will both have Depth = 0 and ChildID = 1, which leads to edge cases if a
```
Comment (Tom94) [inline osu.Framework.Desktop.Tests/Visual/TestCaseContainerState.cs]: This comment needs to be updated, regardless of whether the current PR is kept or whether we revert and add a reference comparion.

File: osu.Framework/Graphics/Containers/CompositeDrawable.cs (hunk 1 above)
Comment (smoogipooo) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs]: I'm not fussed about the atomic, this is not a hot path.

Comment (Tom94) [inline osu.Framework/Graphics/Containers/CompositeDrawable.cs]: That statement was just there to counteract the equally meaningless argument of saving a few bytes. I agree that performance is not the issue here.
